### PR TITLE
[no squash] Some fixes

### PR DESCRIPTION
**Pull request by sfan5**

### CODE REVIEW

File: src/log.cpp [hunk 1]
```diff
@@ -229,45 +227,50 @@ const std::string Logger::getLevelLabel(LogLevel lev)
...
+	const std::string &thread_name = getThreadName();
+	const char *label = getLevelLabel(lev);
 	const std::string timestamp = getTimestamp();
-	std::ostringstream os(std::ios_base::binary);
-	os << timestamp << ": " << label << "[" << thread_name << "]: " << text;
 
-	logToOutputs(lev, os.str(), timestamp, thread_name, text);
+	std::string line = timestamp;
+	line.append(": ").append(label).append("[").append(thread_name)
+		.append("]: ").append(text);
```
Comment (SmallJoker) [inline src/log.cpp:266]: At this point, `snprintf` might be easier to use..?

File: src/log.cpp [hunk 2]
```diff
@@ -229,45 +227,50 @@ const std::string Logger::getLevelLabel(LogLevel lev)
...
+
+	auto it = m_thread_names.find(id);
 	if (it != m_thread_names.end())
 		return it->second;
 
-	std::ostringstream os;
-	os << "#0x" << std::hex << id;
-	return os.str();
+	thread_local std::string fallback_name;
+	if (!fallback_name.empty()) {
```
Comment (grorp) [inline src/log.cpp:247]: isn't this the wrong way around?

Comment (sfan5) [reply src/log.cpp:247]: Ah, yes. I noticed this before actually but forgot to fix it.

File: src/log.cpp (hunk 1 above)
Comment (sfan5) [reply src/log.cpp:266]: true, but that's not C++-y :smiley:

Comment (SmallJoker) [reply src/log.cpp:266]: When you're given a hammer, everything is a nail. Sometimes problems have easier solutions than a hammer but I'll leave this one up to you.

**Pull request merged**